ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -813,6 +813,128 @@ export default {
         },
       },
     },
+    {
+      id: 'E12S Basic Relativity Yellow Hourglass',
+      // Orient where "Yellow" Anger's Hourglass spawns
+      netRegex: NetRegexes.addedCombatantFull({ npcNameId: '9824' }),
+      durationSeconds: 15,
+      infoText: (data, matches, output) => {
+        const y = parseFloat(matches.y) + 75;
+        const x = parseFloat(matches.x);
+
+        // Positions are the 8 cardinals + numerical slop on a radius=20 circle.
+        // Positions are also moved downward 75
+        // N = (0, -95), E = (20, -75), S = (0, -55), W = (-20, -75)
+        // NE = (14, -89), SE = (14, -61), SW = (-14, -61), NW = (-14, -89)
+        // Map N = 0, NE = 1, ..., NW = 7
+        const dirs = {
+          0: output.north(),
+          1: output.northeast(),
+          2: output.east(),
+          3: output.southeast(),
+          4: output.south(),
+          5: output.southwest(),
+          6: output.west(),
+          7: output.northwest(),
+        };
+
+        const dir = Math.round(4 - 4 * Math.atan2(x, y) / Math.PI) % 8;
+
+        return output.hourglass({
+          dir: dirs[dir],
+        });
+      },
+      outputStrings: {
+        north: Outputs.north,
+        northeast: Outputs.northeast,
+        east: Outputs.east,
+        southeast: Outputs.southeast,
+        south: Outputs.south,
+        southwest: Outputs.southwest,
+        west: Outputs.west,
+        northwest: Outputs.northwest,
+        hourglass: {
+          en: 'Yellow: ${dir}',
+        },
+      },
+    },
+    {
+      id: 'E12S Adv Relativity Hourglass Collect',
+      // Collect Sorrow's Hourglass locations
+      netRegex: NetRegexes.addedCombatantFull({ npcNameId: '9823' }),
+      run: (data, matches) => {
+        const id = parseInt(matches.id, 16);
```

Comment:
Is `id` here already base 16? Do you mean `matches.id.toUpperCase()?` (Same question on the other one below.)

Reply:
You're right, it doesn't need to be converted, it'll already be the same.